ecommerce/views/v0/__init__.py

```diff
@@ -84,3 +93,33 @@ def get_queryset(self):
             .select_related("content_type")
             .prefetch_related("purchasable_object")
         )
+
+
+class BasketViewSet(NestedViewSetMixin, ModelViewSet):
```

Comment:
A few things for this view:

- Remove `ModelViewSet` and make it inherit `mixins.ListModelMixin, mixins.RetrieveModelMixin, viewsets.GenericViewSet` instead.
- Add an implementation for `def get_queryset(self, request)` that filters to the current user.
- Set `lookup_field = "user__username"` and `lookup_url_kwarg = "username"` - we'll use `username` as the identifier instead of `Basket.id` because each user has only 1 basket and username is a value the frontend already knows.
- Add an implementation for `def get_object(self,request, username=None)` that does a `Basket.objects.get_or_create(user=request.user)` - we're not using the `queryset` here intentionally